submissions/Roophee/pair-game/index.js

```diff
@@ -0,0 +1,173 @@
+"use strict";
+
+const heroesArray = ['luke', 'vader', 'solo', 'ob1', 'palpatine', 'leia', 'r2d2-c3po', 'chewbacca', 'bad1', 'bad2', 'bad3', 'bad4', 'good1', 'good3', 'good4', 'robot1', 'yoda'],
+    indexInUse = [],
+    imagesCountForTwo = {},
+    prePair = [],
+    pairs = [],
+    gameBg = document.querySelector('.game'),
+    cardPlace = document.querySelector('.game__card-place'),
+    audioItem = document.querySelector('audio'),
+    audioButton = document.querySelector('.audio__button'),
+    quantityOfUniqueCards = 17,
+    quantityOfCardsOnTheScreen = 12,
+    maxQuantityOfPairsOnTheScreen = 6;
+
+
+// TAKE RANDOM CARDS FROM POOL OF CARDS
+const cardsForPlay = function () {
+    while (true) {
+        let randomIndex = Math.floor(Math.random() * quantityOfUniqueCards);
+        if (indexInUse.length === maxQuantityOfPairsOnTheScreen) {
+            break
+        } else if (indexInUse.indexOf(randomIndex) < 0) {
+            indexInUse.push(randomIndex);
+        }
+    }
+}
+
+cardsForPlay();
+
+//COUNT HOW MANY SAME IMAGES ADD IN GAME
+const imageCounter = function (indx) {
+    if (imagesCountForTwo.hasOwnProperty(indx) && imagesCountForTwo[indx] >= 2) {
+        return -2;
+    } else if (imagesCountForTwo.hasOwnProperty(indx) && imagesCountForTwo[indx] < 2) {
+        imagesCountForTwo[indx] += 1;
+        return indx;
+    } else if (!imageCounter.hasOwnProperty(indx)) {
+        imagesCountForTwo[indx] = 1;
+        return indx;
+    }
+}
+
+// TAKE RANDOM IMAGE FROM POOL OF IMAGES FOR THIS GAME
+const getRandomImage = function () {
+    while (true) {
+        let randomIndex = Math.floor(Math.random() * maxQuantityOfPairsOnTheScreen);
+        if (imageCounter(randomIndex) >= -1) {
+            return randomIndex;
+        }
+    }
+}
+
+    //MAKE ANIMATION ON PAGE DOWNLOAD
+const animateOnStart = function(){
+    gameBg.style.top = `${-gameBg.clientHeight - gameBg.offsetTop}px`;
+    cardPlace.style.bottom = `${-document.documentElement.clientHeight * 2}px`;
+    setTimeout(function (){
+        gameBg.style.top = '0px';
+        cardPlace.style.bottom = '0px';
+    },500)
+}
+
+document.addEventListener('DOMContentLoaded', animateOnStart);
+
+// ADD CARDS ON DOCUMENT
+let addCardsOnPage = function () {
+    let cardsContainerOnDynamicGeneration = document.createDocumentFragment();
+    for (let i = 0; i < quantityOfCardsOnTheScreen; i++) {
+        let cardNameInsertOnDynamicGeneration = heroesArray[indexInUse[getRandomImage()]],
+            cardItem = document.createElement('div');
+        cardItem.classList.add('game__card-holder');
+        cardItem.innerHTML = `<div class="game__card-item">
+                            <div class="game__card-front">
+                                <img src="img/${cardNameInsertOnDynamicGeneration}.jpg" alt="${cardNameInsertOnDynamicGeneration}"  class="game__card-img">
```

Comment:
store full src in your object/array etc.

Reply:
If I will use your approach, I will  need split string to get only name of card to insert it in `alt` atribute. Is it realy necessary?
In my approach, I take name of card from array and then I can handly insert one string twice in `src` and `alt`.

Reply:
What you will do if one image will be in `.jpeg` format another in `.jpg` and others in `.png` ? (for example)
You don't need split string. It is better to store image info in object like this 
```
{
  alt: 'image description',
  src: 'path',
  // other properties: id, text, etc.
}
```

Reply:
@Roophee

Reply:
In my implementation of MPG, I have only `.jpeg` images and I have not need to make this check.
In another  situation this check ( `.png`  or  `.jpeg`) may have a sense.

Store info in object  - is clever, I will use this approach in future. Thanks.

Reply:
@babayK0  done. Please, review.